heron/common/src/cpp/metrics/metricsmgr-client.cpp

```diff
@@ -63,6 +63,12 @@ void MetricsMgrClient::HandleClose(NetworkErrorCode) {
 
 void MetricsMgrClient::ReConnect() { Start(); }
 
+void MetricsMgrClient::SetPublisherPort(const sp_int32 _port) {
+  if (0 == port_ && 0 != _port) {
```

Comment:
@huijunw If you can hold on for a week or so, I have other refactoring coming in stmgr that hopefully should alleviate some of this